pkg/virt-controller/watch/vmi.go

```diff
@@ -2341,3 +2357,43 @@ func generateInterfaceStatusPatchRequest(oldInterfaceStatus []byte, newInterface
 		fmt.Sprintf(`{ "op": "add", "path": "/status/interfaces", "value": %s }`, string(newInterfaceStatus)),
 	}
 }
+
+func (c *VMIController) isCPUHotplugReady(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) (bool, error) {
+	vmiResourceRenderer, err := services.NewVMIResourceRenderer(vmi, nil, c.clusterConfig, c.clientset)
+	if err != nil {
+		return false, err
+	}
+
+	computeContainer := GetContainerOfPod(pod, "compute")
+	if computeContainer == nil {
+		return false, fmt.Errorf("could not find compute container")
+	}
+
+	vmiResourceRequests := vmiResourceRenderer.ResourceRequirements().Requests
+	ccResourceRequests := computeContainer.Resources.Requests
+
+	if vmiResourceRequests.Cpu().Value() != ccResourceRequests.Cpu().Value() ||
```

Comment:
Is this resilient against upgrades(different versions having different calculations) and changes to Kubevirt/ cluster config?